src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedData: GroupsMap<T> = {};
+  const allKeysValues: Array<string> = [];
+
+  for (const item of items) {
+    const value = `${item[key]}`;
```

Comment:
Directly converting the key to a string might cause issues if the key is a symbol or if the object has a custom toString method that doesn't return a unique value. Ensure that the key can be safely converted to a string that uniquely identifies the property.